### PR TITLE
Added CODEOWNERS file with list of names

### PR DESCRIPTION
**Pull request by VladimirKravtsov**
### What changes were proposed in this pull request?
Introduce CODEOWNERS file with the list of entire team members' names.

### Why are the changes needed?
The folks mentioned in the file will be marked as code owners in GitHub UI. And what more important:
>Code owners are automatically requested for review when someone opens a pull request that modifies code that they own. Code owners are not automatically requested to review draft pull requests. For more information about draft pull requests, see "About pull requests."

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# These owners will be the default owners for everything in
+# the repo. Unless a later match takes precedence,
```
Comment (farhan5900) [inline .github/CODEOWNERS]: The comment `Unless a later match takes precedence` is not clear. I think it should be `When several patterns match the same file, the last one will take precedence`.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+# @global-owner1 and @global-owner2 will be requested for
+# review when someone opens a pull request.
```
[the remark this answers is not in the document]
Comment (samvantran) [reply .github/CODEOWNERS]: Copy-pasting only a subsection of the [Github examples](https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners) makes this doc a little confusing. I think it would be clearer if the entire file read like this:
```
# The following list of usernames are owners of this project 
# and will tagged as reviewers when you create a PR:

*       @VladimirKravtsov @akirillov @rpalaznik @farhan5900 @samvantran @alembiewski

# To add more rules, see: https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners

[the remark this answers is not in the document]
Comment (VladimirKravtsov) [reply .github/CODEOWNERS]: > * sometimes it is sufficient to request a review from just a single person when the change is related to some specific code area developed by that person

 To cover that case,  it is possible to add to the file a more specific mapping like ’/kudo-spark @akirillov’. I think when responsibilities for the operator pieces became persistent across team members, we can easily add such mappings. 

> * spamming the whole dev team every time a PR is created can lead to a situation when the actual requests are ignored because "anyway, the whole team received the same request."

I walked through across the last 10 - 15 Rs in this repo, and 90% of them have the entire pillar team in reviewers. It means having current CODEOWNERS, in most cases, will no need to adjust the reviewers list.
 
> TL;DR requesting a review on per-PR basis doesn't seem to be an issue and works for all other repos so far.

Yes, it is. 

the codeowners configuration maybe bring some additional convenience into the development process, but perhaps not if so, we can easily remove it.

**Comment by akirillov** [comment]
+1 to what Sam said:
> However, one caveat: if this automates the tagging of reviewers during PR creation time but we agree to deselect reviewers if it doesn't apply to them before we click 'submit' on PRs, then I'm 100% for this bc it helps to save the clicks of searching for reviewers every time.

**Comment by VladimirKravtsov** [comment]
I have added specific folders to usernames mapping. I'm not sure that is the best one, so feel free to suggest a better distribution.

**Comment by samvantran** [comment]
Thanks @VladimirKravtsov - I was fine with having a blanket list of reviewers as I suggested [here](https://github.com/mesosphere/kudo-spark-operator/pull/37#discussion_r338313235) but up to you guys.

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# The following list of usernames are owners of this project
+# and will tagged as reviewers when you create a PR:
```
Comment (samvantran) [inline .github/CODEOWNERS]:
```suggestion
# and will be tagged as reviewers when you create a PR:
```
sorry I had a typo in my earlier suggestion
